packages/provider/src/RelayClient.ts

```diff
@@ -579,9 +594,13 @@ export class RelayClient {
 
   async _resolveDependencies ({
     provider,
-    config = {},
+    config,
```

Comment:
Are we sure that we don't depend anywhere on this default value that you removed?

Reply:
Well, it is unlikely as the default value here is literally an empty object. However, I've changed it to accept the fully initialized `GSNConfig` because the `_resolveDependencies` is only called after the `_resolveConfiguration`.
I don't see how this empty config could ever be used here at all.